### ISSUE TITLE
decoder with open boundary

### ISSUE THREAD

**Issue by a0024356** [chaeyeunpark/UnionFind#14]
Hi, I am a beginner at QEC and I would like to use this code. But the explanation seems to assume the periodic boundary. Could you tell me if this project also supports union-find decoder with open boundary?

**Comment by chaeyeunpark** [comment]
Yes, the current version of the code supports a QEC with the open boundary condition. You can just pass the corresponding parity matrix as in this [usage document](https://unionfind.readthedocs.io/en/latest/usage.html). If you want faster runtime, you may need to use C++ code directly, which I explained [here](https://unionfind.readthedocs.io/en/latest/custom_lattice.html).

**Comment by a0024356** [comment]
Thank you for your quick reply, but I got a segmentation fault error when I used a parity matrix like this:
H=[[1 1 0 1 0 0 0 0 0 0 0 0 0]
      [0 1 1 0 1 0 0 0 0 0 0 0 0]
      [0 0 0 1 0 1 1 0 1 0 0 0 0]
      [0 0 0 0 1 0 1 1 0 1 0 0 0]
      [0 0 0 0 0 0 0 0 1 0 1 1 0]
      [0 0 0 0 0 0 0 0 0 1 0 1 1]]
I am trying to decode d=3 surface code with the open boundary.
Is there any good solution? Or isn't it the right way to create parity matrix?

**Comment by chaeyeunpark** [comment]
Hi, I confirm that the current implementation does not support this parity matrix as it assumes that a qubit is associated with two nearby syndromes. I am fixing this by relaxing the condition. I will update here when it is fixed. Thanks for reporting an issue :smile:

**Comment by a0024356** [comment]
Thank you for the verification. I look forward to hearing from you.
